### ISSUE TITLE
`file:` protocol in `package-lock.json` seems to not be supported

### ISSUE THREAD

**Issue by wooorm** [mbalabash/sdc-check#1]
Hey! Cool project :) I tried it out and got some weird errors. Inside [`mdx-js/mdx`](https://github.com/mdx-js/mdx):

```sh
node -v # v17.4.0
npm -v # 8.5.5
npx sdc-check -d .
```

Yields:

```
Errors: 22
[
  {
    "metric": "lockfile-is-not-safe",
    "message": "detected invalid host(s) for package: @mdx-js/esbuild@file:packages/esbuild-ebc0e1c9cb0044a8d3631c422ddb2f0182111f49\n    expected: registry.npmjs.org\n    actual: \n",
    "package": "@mdx-js/esbuild@file:packages/esbuild-ebc0e1c9cb0044a8d3631c422ddb2f0182111f49"
  },
  {
    "metric": "lockfile-is-not-safe",
    "message": "detected invalid host(s) for package: @mdx-js/loader@file:packages/loader-bd54eca448ffeb221a8b218aea8e32061618c046\n    expected: registry.npmjs.org\n    actual: \n",
    "package": "@mdx-js/loader@file:packages/loader-bd54eca448ffeb221a8b218aea8e32061618c046"
  },
  ...,
  {
    "metric": "package-is-too-new",
    "package": "@mdx-js/preact@2.1.1",
    "message": "package release date is 2022-03-31"
  },
  {
    "metric": "package-is-too-new",
    "package": "@mdx-js/mdx@2.1.1",
    "message": "package release date is 2022-03-31"
  },
```

So, it seems npm workspaces, or at least local references, don’t work well?

**Comment by mbalabash** [comment]
Hey @wooorm 
I'll take a look at it. It may be a bug in [lockfile-lint](https://github.com/lirantal/lockfile-lint) or something wrong with how I have used it in `sdc-check`. I will come back with new info soon

**Comment by mbalabash** [comment]
I found the answer in `lockfile-lint` [issue #23](https://github.com/lirantal/lockfile-lint/issues/23). 
Fix was applied and released in `sdc-check@1.0.1`.
